client/app/shared/ci-helper/ci-helper.controller.js

```diff
@@ -274,13 +274,19 @@ const CiHelperController = function CiHelperController(
                 let provider = vm.providers[0];
                 //is config
                 if (launcher.hasOwnProperty('parentLauncherId')) {
-                    const integration = vm.integrations.find(({ id }) => id === launcher.providerId);
-                    const predefinedProvider = vm.providers.find(({ name }) => name.toLowerCase() === integration.name.toLowerCase());
-
-                    if (predefinedProvider) {
-                        provider = predefinedProvider;
+                    if (launcher.hasOwnProperty('providerId')) {
+                        const integration = vm.integrations.find(({ id }) => id === launcher.providerId);
+                        const predefinedProvider = vm.providers.find(({ name }) => name.toLowerCase() === integration.name?.toLowerCase());
```

Comment:
`integration` can be `undefined`. In this case, you'll get an exception. Also, in this case, you don't need to iterate providers array at all.

Reply:
fixed